src/hooks/use-get-user-tranche-balances.ts

```diff
@@ -7,25 +7,32 @@ import {
   useAppState,
 } from "../contexts/app-state/app-state-context";
 import { BigNumber } from "../lib/bignumber";
+import { useTranches } from "./use-tranches";
 
 export const useGetUserTrancheBalances = (
   address: string,
   vesting: VegaVesting
 ) => {
   const { appDispatch } = useAppState();
+  const { tranches } = useTranches();
+
   return React.useCallback(async () => {
     appDispatch({
       type: AppStateActionType.SET_TRANCHE_ERROR,
       error: null,
     });
+
     try {
-      const tranches = await vesting.getAllTranches();
-      const userTranches = tranches.filter((t) =>
-        t.users.some(
-          ({ address: a }) =>
-            a && address && a.toLowerCase() === address.toLowerCase()
-        )
-      );
+      if (!tranches!) {
+        return;
+      }
+      const userTranches =
+        tranches.filter((t) =>
+          t.users.some(
+            ({ address: a }) =>
+              a && address && a.toLowerCase() === address.toLowerCase()
```

Comment:
(question): I don't understand this code - with `address` being renamed as `a`, I think this will error saying that address is undefined. Does `userTranches` ever return anything other than an empty array?